package com.samsung.telephony.sysprop;

import android.os.SystemProperties;
import java.util.ArrayList;
import java.util.Iterator;
import java.util.List;
import java.util.Locale;
import java.util.Optional;
import java.util.StringJoiner;
import java.util.function.Function;

/* loaded from: classes6.dex */
public final class SemTelephonyVendorProps {
    private SemTelephonyVendorProps() {
    }

    private static Boolean tryParseBoolean(String str) {
        if (str == null) {
            return null;
        }
        String lowerCase = str.toLowerCase(Locale.US);
        lowerCase.hashCode();
        switch (lowerCase) {
        }
        return null;
    }

    private static Integer tryParseInteger(String str) {
        try {
            return Integer.valueOf(str);
        } catch (NumberFormatException unused) {
            return null;
        }
    }

    private static Integer tryParseUInt(String str) {
        try {
            return Integer.valueOf(Integer.parseUnsignedInt(str));
        } catch (NumberFormatException unused) {
            return null;
        }
    }

    private static Long tryParseLong(String str) {
        try {
            return Long.valueOf(str);
        } catch (NumberFormatException unused) {
            return null;
        }
    }

    private static Long tryParseULong(String str) {
        try {
            return Long.valueOf(Long.parseUnsignedLong(str));
        } catch (NumberFormatException unused) {
            return null;
        }
    }

    private static Double tryParseDouble(String str) {
        try {
            return Double.valueOf(str);
        } catch (NumberFormatException unused) {
            return null;
        }
    }

    private static String tryParseString(String str) {
        if ("".equals(str)) {
            return null;
        }
        return str;
    }

    private static <T extends Enum<T>> T tryParseEnum(Class<T> cls, String str) {
        try {
            return (T) Enum.valueOf(cls, str.toUpperCase(Locale.US));
        } catch (IllegalArgumentException unused) {
            return null;
        }
    }

    private static <T> List<T> tryParseList(Function<String, T> function, String str) {
        if ("".equals(str)) {
            return new ArrayList();
        }
        ArrayList arrayList = new ArrayList();
        int i = 0;
        while (true) {
            StringBuilder sb = new StringBuilder();
            while (i < str.length() && str.charAt(i) != ',') {
                if (str.charAt(i) == '\\') {
                    i++;
                }
                if (i == str.length()) {
                    break;
                }
                sb.append(str.charAt(i));
                i++;
            }
            arrayList.add(function.apply(sb.toString()));
            if (i == str.length()) {
                return arrayList;
            }
            i++;
        }
    }

    private static <T extends Enum<T>> List<T> tryParseEnumList(Class<T> cls, String str) {
        if ("".equals(str)) {
            return new ArrayList();
        }
        ArrayList arrayList = new ArrayList();
        for (String str2 : str.split(",")) {
            arrayList.add(tryParseEnum(cls, str2));
        }
        return arrayList;
    }

    private static String escape(String str) {
        return str.replaceAll("([\\\\,])", "\\\\$1");
    }

    private static <T> String formatList(List<T> list) {
        StringJoiner stringJoiner = new StringJoiner(",");
        Iterator<T> it = list.iterator();
        while (it.hasNext()) {
            T next = it.next();
            stringJoiner.add(next == null ? "" : escape(next.toString()));
        }
        return stringJoiner.toString();
    }

    private static String formatUIntList(List<Integer> list) {
        StringJoiner stringJoiner = new StringJoiner(",");
        Iterator<Integer> it = list.iterator();
        while (it.hasNext()) {
            Integer next = it.next();
            stringJoiner.add(next == null ? "" : escape(Integer.toUnsignedString(next.intValue())));
        }
        return stringJoiner.toString();
    }

    private static String formatULongList(List<Long> list) {
        StringJoiner stringJoiner = new StringJoiner(",");
        Iterator<Long> it = list.iterator();
        while (it.hasNext()) {
            Long next = it.next();
            stringJoiner.add(next == null ? "" : escape(Long.toUnsignedString(next.longValue())));
        }
        return stringJoiner.toString();
    }

    private static <T extends Enum<T>> String formatEnumList(List<T> list, Function<T, String> function) {
        StringJoiner stringJoiner = new StringJoiner(",");
        Iterator<T> it = list.iterator();
        while (it.hasNext()) {
            T next = it.next();
            stringJoiner.add(next == null ? "" : function.apply(next));
        }
        return stringJoiner.toString();
    }

    public static Optional<String> debug_level() {
        return Optional.ofNullable(tryParseString(SystemProperties.get("ro.vendor.boot.debug_level")));
    }

    public static Optional<String> cp_debug_level() {
        return Optional.ofNullable(tryParseString(SystemProperties.get("ro.vendor.boot.cp_debug_level")));
    }

    public static Optional<String> force_upload() {
        return Optional.ofNullable(tryParseString(SystemProperties.get("ro.vendor.boot.force_upload")));
    }

    public static Optional<String> build_characteristics() {
        return Optional.ofNullable(tryParseString(SystemProperties.get("ro.vendor.build.characteristics")));
    }

    public static Optional<Integer> radio_default_network() {
        return Optional.ofNullable(tryParseInteger(SystemProperties.get("ro.vendor.radio.default_network")));
    }

    public static void radio_default_network(Integer num) {
        SystemProperties.set("ro.vendor.radio.default_network", num == null ? "" : num.toString());
    }

    public static Optional<String> radio_default_network_by() {
        return Optional.ofNullable(tryParseString(SystemProperties.get("ro.vendor.radio.default_network_by")));
    }

    public static void radio_default_network_by(String str) {
        SystemProperties.set("ro.vendor.radio.default_network_by", str == null ? "" : str.toString());
    }

    public static Optional<String> sec_radio_def_network() {
        return Optional.ofNullable(tryParseString(SystemProperties.get("ro.vendor.sec.radio.def_network")));
    }

    public static void sec_radio_def_network(String str) {
        SystemProperties.set("ro.vendor.sec.radio.def_network", str == null ? "" : str.toString());
    }

    public static Optional<Integer> vendor_api_level() {
        return Optional.ofNullable(tryParseInteger(SystemProperties.get("ro.vendor.api_level")));
    }
}
